solana/ts/sdk/utils.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  Layout,
+  CustomConversion,
+  encoding,
+} from "@wormhole-foundation/sdk-base";
+
+import { PublicKey, PublicKeyInitData } from "@solana/web3.js";
+import { BN } from "@coral-xyz/anchor";
+
+const CHAIN_ID_BYTE_SIZE = 2;
+export const chainIdToBeBytes = chainId => encoding.bignum.toBytes(chainId, CHAIN_ID_BYTE_SIZE);
+
+export const BPF_LOADER_UPGRADEABLE_PROGRAM_ID =
+  new PublicKey("BPFLoaderUpgradeab1e11111111111111111111111");
+
+export function programDataAddress(programId: PublicKeyInitData) {
+  return PublicKey.findProgramAddressSync(
+      [new PublicKey(programId).toBytes()],
+      BPF_LOADER_UPGRADEABLE_PROGRAM_ID,
+  )[0];
+}
+
+export const pubKeyConversion = {
+  to:  (encoded: Uint8Array) => new PublicKey(encoded),
+  from: (decoded: PublicKey) => decoded.toBytes(),
+} as const satisfies CustomConversion<Uint8Array, PublicKey>;
+
+//neither anchor nor solana web3 have a built-in way to parse this, because ofc they don't
+export const programDataLayout = [
+  { name: "slot", binary: "uint", endianness: "little", size: 8 },
+  { name: "upgradeAuthority", binary: "switch", idSize: 1, idTag: "isSome", layouts: [
+      [[0, false], []],
+      [[1, true], [{name: "value", binary: "bytes", size: 32, custom: pubKeyConversion}]],
+    ],
+  }
+] as const satisfies Layout;
+
+export const U64 = {
+  MAX: new BN((2n**64n - 1n).toString()),
```

Comment:
CI tests seem to be failing around here. I'm not sure exactly which ES version is used by the CI pipeline.

@evan-gray might be able to shed some light here? Otherwise I'll look around and see whether I can track down the issue. It looks like we'll need to update the version used, or else find a different BigInt solution here.

```
ntt-ci-tests │ TSError: ⨯ Unable to compile TypeScript:
 ntt-ci-tests │ solana_binding/ts/sdk/utils.ts(39,16): error TS2737: BigInt literals are not available when targeting lower than ES2020.
 ntt-ci-tests │ solana_binding/ts/sdk/utils.ts(39,16): error TS2791: Exponentiation cannot be performed on 'bigint' values unless the 'target' option is set to 'es2016' or later.
 ntt-ci-tests │ solana_binding/ts/sdk/utils.ts(39,20): error TS2737: BigInt literals are not available when targeting lower than ES2020.
 ntt-ci-tests │ solana_binding/ts/sdk/utils.ts(39,26): error TS2737: BigInt literals are not available when targeting lower than ES2020.
 ```

Reply:
Ah I think the issue is in the modifications in `tsconfig.json`. These need to be updated within CI, or else you'll need to use methods compatible with the old version and revert `tsconfig.json`